website/client-old/js/controllers/hallCtrl.js

```diff
@@ -1,16 +1,56 @@
 "use strict";
 
-habitrpg.controller("HallHeroesCtrl", ['$scope', '$rootScope', 'User', 'Notification', 'ApiUrl', 'Hall',
-  function($scope, $rootScope, User, Notification, ApiUrl, Hall) {
+habitrpg.controller("HallHeroesCtrl", ['$scope', '$rootScope', 'User', 'Notification', 'ApiUrl', 'Hall', 'Content',
+  function($scope, $rootScope, User, Notification, ApiUrl, Hall, Content) {
     $scope.hero = undefined;
     $scope.currentHeroIndex = undefined;
     $scope.heroes = [];
+    $scope.allItemPaths = getAllItemPaths();
 
     Hall.getHeroes()
       .then(function (response) {
         $scope.heroes = response.data.data;
       });
 
+    function _getFormattedItemReference (pathPrefix, keys, values) {
+      var finishedString = '\n'.concat('path: ', pathPrefix, ', ', 'value: {', values, '}\n');
+
+      _.each(keys, (key) => {
+        finishedString = finishedString.concat('\t', pathPrefix, '.', key, '\n');
+      });
+      return finishedString;
+    }
+
+    function getAllItemPaths () {
+      var content = Content;
+
+      var quests = _getFormattedItemReference('items.quests', _.keys(content.quests), 'Numeric Quantity');
+      var mounts = _getFormattedItemReference('items.mounts', _.keys(content.mountInfo), 'Boolean');
+      var food = _getFormattedItemReference('items.food', _.keys(content.food), 'Numeric Quantity');
+      var eggs = _getFormattedItemReference('items.eggs', _.keys(content.eggs), 'Numeric Quantity');
+      var hatchingPotions = _getFormattedItemReference('items.hatchingPotions', _.keys(content.hatchingPotions), 'Numeric Quantity');
+      var pets = _getFormattedItemReference('items.pets', _.keys(content.petInfo), '-1: Owns Mount, 0: Not Owned, 1-49: Progress to mount');
+      var special = _getFormattedItemReference('items.special', _.keys(content.special), 'Numeric Quantity');
+
+      var mystery = _.flatten(_.keys(content.mystery).map((mysterySetKey) => {
```

Comment:
You don't need to add the mystery sets here. The gear items are already included in the gear content object.